_posts/2018-10-02-auto-tune-all.md

```diff
@@ -0,0 +1,198 @@
+---
+layout: post
+title: Automatic Kernel Optimization for Deep Learning on All Hardware Platforms
+date: 2018-10-02
+author: Lianmin Zheng, Eddie Yan
+tags:
+  - tvm
+---
+
+How to optimize the performance of deep neural network on a diverse range of hardware platforms is still a hard
+problem for AI developers. In terms of system support, we are facing a many-to-many problem here:
+deploying trained models from multiple frontends (e.g. Tensorflow, ONNX, MXNet) to multiple
+hardware platforms (e.g. CPU, GPU, Accelerators). On the most performance critical part of
+this problem is how to get high performance kernel implementation for growing model
+architectures and hardware platforms.
+
+To address this challenge, TVM takes a full stack compiler approach. Combining code generator and auto-tuner in TVM,
+we are able to generate kernels that are comparable to heavily hand-optimized libraries,
+and obtain the state-of-the-art inference performance on hardware platforms including
+ARM CPUs, Intel CPUs, Mali GPUs, NVIIDA GPUs and AMD GPUs.
+
+In this blog post, I will show the workflow of automatic kernel optimization in TVM compiler stack and 
+benchmark results on several hardware platforms.
+
+# System Overview
+
+{:center: style="text-align: center"}
+![image](/images/autotune-all/circle.png){: width="35%"}
+{:center}
+<center> Figure 1. System Overview </center> <p></p>
+
+Kernel optimization in TVM is done in an iterative loop fashion.
+As shown in Figure 1, the automatic kernel optimization takes a neural network (typically in computational graph representation)
+from frontend frameworks as input, and generates kernels for all the operators in this network.
+
+The inner loop consists of scalable RPC runtime, machine learning based tuners and a tensor compiler.
```

Comment:
The inner loop uses a scalable RPC runtime, machine learning-based tuners, and a tensor compiler.